src/ucs/arch/rv64/bitops.h

```diff
@@ -0,0 +1,41 @@
+/**
+* Copyright (C) Tactical Computing Labs, LLC. 2022. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCS_ARCH_RV64_BITOPS_H_
+#define UCS_ARCH_RV64_BITOPS_H_
+
+#include <ucs/sys/compiler_def.h>
+#include <stdint.h>
+
+static UCS_F_ALWAYS_INLINE unsigned __ucs_ilog2_u32(uint32_t n)
+{
+    return 31 - __builtin_clz(n);
+}
+
+static UCS_F_ALWAYS_INLINE unsigned __ucs_ilog2_u64(uint64_t n)
+{
+    uint32_t lower, upper;
+
+    upper = n >> 32;
+    lower = n;
+    if (upper == 0) {
+        return __ucs_ilog2_u32(lower);
+    } else {
+        return 32 + __ucs_ilog2_u32(upper);
```

Comment:
does RISC-V have CLZ for 64-bit?

Reply:
I discovered that there is a `__builtin_clzll` which always 8 bytes - I switched to that and I switched the ffs fuctions to builtin equivalents. This means that if the hardware + compiler supports the `Zbb` extension (which adds clz instructions) then it can used optimised versions - otherwise it will fallback to a software implementation.